oracle/oracle.go

```diff
@@ -0,0 +1,84 @@
+package oracle
+
+import (
+	"github.com/algorand/go-algorand-sdk/stateproofs/stateproofverification"
+	"github.com/algorand/go-algorand-sdk/stateproofs/transactionverificationtypes"
+	"github.com/algorand/go-algorand-sdk/types"
+)
+
+// Oracle is in charge of ingesting state proofs, in a chain, and saving their block interval attesting
```

Comment:
nit: "Oracle is responsible for ingesting State Proofs in chronological order and saving their block interval commitments to form a window of verified Algorand history"